### ISSUE TITLE
Support querying by major API version

### ISSUE THREAD

**Issue by Katrix** [Minecrell/SpongeDownloads#6]
Calling this you would expect to get API 6 versions back, but instead you get (currently) 7.1.0 and 8.0.0 versions instead.
https://dl-api.spongepowered.org/v1/org.spongepowered/spongevanilla/downloads?version=6

**Comment by ryantheleach** [comment]
Hey guys, just a heads up this is blocking SpongeStart(V2) from using the correct snapshot versions when debugging, so is causing some pain for newbie plugin developers that don't have a full SF/SV debug environment.

So People would really really appreciate seeing a fix.

**Comment by stephan-gh** [comment]
The version parameter works as expected, it queries the implementation version, not the API version.

The correct parameter here is `?spongeapi=` but currently that doesn't support adding only the major version.

**Comment by ryantheleach** [comment]
`https://dl-api.spongepowered.org/v1/org.spongepowered/spongevanilla/downloads?spongeapi=7.0.0` results in no results.

**Comment by stephan-gh** [comment]
@ryantheleach Correct, because the implementation always targets a snapshot version of the API.

Example would be https://dl-api.spongepowered.org/v1/org.spongepowered/spongevanilla/downloads?spongeapi=7.1.0-SNAPSHOT-dfc5c4a7

Granted, this isn't very useful but back when I designed it I mostly had only the downloads page in mind. I will add something for this when I have time.

**Comment by progwml6** [comment]
this support should be added to other relevent filters as well, such as the one for recommended version
